File: Question 0958 - Check Completeness of a Binary Tree/src/Solution.java
class Solution {
    private int countNodes(TreeNode root)
    {
        if(root == null) return 0;

        return 1+countNodes(root.left) + countNodes(root.right);
    }

    private boolean isCBT(TreeNode root, int index, int count)
    {
        if(root == null) return true;

        if(index >= count) return false;
        else
        {
            boolean left = isCBT(root.left, 2*index + 1, count);
            boolean right = isCBT(root.right, 2*index + 2, count);
            return left && right;
        }
    }
    public boolean isCompleteTree(TreeNode root) {
        int index = 0;
        int totalCount = countNodes(root);
        return isCBT(root, index, totalCount);
    }
}

File: Question 0958 - Check Completeness of a Binary Tree/src/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution
{
public:
    bool isCompleteTree(TreeNode *root)
    {
        if (!root)
            return true;

        queue<TreeNode *> q;
        q.push(root);
        bool founNull = false;

        while (!q.empty())
        {
            TreeNode *curr = q.front();
            q.pop();

            if (!curr)
                founNull = true;
            else
            {
                if (founNull)
                    return false;

                q.push(curr->left);
                q.push(curr->right);
            }
        }

        return true;
    }
};